File: projetoES3/src/model/domain/Endereco.java
package model.domain;

public class Endereco extends EntidadeDominio{

	private String nomeEnd;
	private String tipoEnd;
	private String tipoLogr;
	private String tipoResid;
	private String logradouroEnd;
	private int numeroEnd;
	private String bairroEnd;
	private String cepEnd;
	private Cidade cidadeEnd;
	private String observacoesEnd;
	//private int fornecedorId;

	public Endereco(String nomeEnd, String tipoEnd, String tipoLogr, 
			String tipoResid, String logradouroEnd, int numeroEnd, String bairroEnd,
			String cepEnd, Cidade cidadeEnd, String observacoesEnd) {
		this.nomeEnd = nomeEnd;
		this.tipoEnd = tipoEnd;
		this.tipoLogr = tipoLogr;
		this.tipoResid = tipoResid;
		this.logradouroEnd = logradouroEnd;
		this.numeroEnd = numeroEnd;
		this.bairroEnd = bairroEnd;
		this.cepEnd = cepEnd;
		this.cidadeEnd  = cidadeEnd;
		this.observacoesEnd = observacoesEnd;
//		this.fornecedorId = fornecedorId;
	}

	
	public boolean validarEndereco() {
		System.out.println("ENDEREÇO OK!");
		return true;
	}
	
	public String getNomeEnd() {
		return nomeEnd;
	}

	public void setNomeEnd(String nomeEnd) {
		this.nomeEnd = nomeEnd;
	}

	public String getTipoEnd() {
		return tipoEnd;
	}

	public void setTipoEnd(String tipoEnd) {
		this.tipoEnd = tipoEnd;
	}

	public String getTipoLogr() {
		return tipoLogr;
	}

	public void setTipoLogr(String tipoLogr) {
		this.tipoLogr = tipoLogr;
	}

	public String getTipoResid() {
		return tipoResid;
	}

	public void setTipoResid(String tipoResid) {
		this.tipoResid = tipoResid;
	}

	public String getLogradouroEnd() {
		return logradouroEnd;
	}

	public void setLogradouroEnd(String logradouroEnd) {
		this.logradouroEnd = logradouroEnd;
	}

	public int getNumeroEnd() {
		return numeroEnd;
	}

	public void setNumeroEnd(int numeroEnd) {
		this.numeroEnd = numeroEnd;
	}

	public String getBairroEnd() {
		return bairroEnd;
	}

	public void setBairroEnd(String bairroEnd) {
		this.bairroEnd = bairroEnd;
	}

	public String getCepEnd() {
		return cepEnd;
	}

	public void setCepEnd(String cepEnd) {
		this.cepEnd = cepEnd;
	}

	public Cidade getCidadeEnd() {
		return cidadeEnd;
	}

	public void setCidadeEnd(Cidade cidadeEnd) {
		this.cidadeEnd = cidadeEnd;
	}
	
	public String getObservacoesEnd() {
		return observacoesEnd;
	}

	public void setObservacoesEnd(String observacoesEnd) {
		this.observacoesEnd = observacoesEnd;
	}

//
//	public int getFornecedorId() {
//		return fornecedorId;
//	}
//
//
//	public void setFornecedorId(int fornecedorId) {
//		this.fornecedorId = fornecedorId;
//	}

	
	
	
}

File: typeScript/src/model/domain/Endereco.ts
import {EntidadeDominio} from "./EntidadeDominio";
import {Cidade} from "./Cidade";
import {Fornecedor} from "./Fornecedor";

export class Endereco extends EntidadeDominio {
    constructor(
        private nomeEnd: string,
        private  tipoEnd: string,
        private  tipoLogr: string,
        private  tipoResid: string,
        private  logradouroEnd: string,
        private  numeroEnd: number,
        private  bairroEnd: string,
        private  cepEnd: string,
        private  cidadeEnd: number,
        private observacoesEnd: string,
    ) {
        super();
    }

    public getNomeEnd(): string {
        return this.nomeEnd;
    }

    public setNomeEnd(nomeEnd: string): void {
        this.nomeEnd = nomeEnd;
    }

    public getTipoEnd(): string {
        return this.tipoEnd;
    }

    public setTipoEnd(tipoEnd: string): void {
        this.tipoEnd = tipoEnd;
    }

    public getTipoLogr(): string {
        return this.tipoLogr;
    }

    public setTipoLogr(tipoLogr: string): void {
        this.tipoLogr = tipoLogr;
    }

    public getTipoResid(): string {
        return this.tipoResid;
    }

    public setTipoResid(tipoResid: string): void {
        this.tipoResid = tipoResid;
    }

    public getLogradouroEnd(): string {
        return this.logradouroEnd;
    }

    public setLogradouroEnd(logradouroEnd: string): void {
        this.logradouroEnd = logradouroEnd;
    }

    public getNumeroEnd(): number {
        return this.numeroEnd;
    }

    public setNumeroEnd(numeroEnd: number): void {
        this.numeroEnd = numeroEnd;
    }

    public getBairroEnd(): string {
        return this.bairroEnd;
    }

    public setBairroEnd(bairroEnd: string): void {
        this.bairroEnd = bairroEnd;
    }

    public getCepEnd(): string {
        return this.cepEnd;
    }

    public setCepEnd(cepEnd: string): void {
        this.cepEnd = cepEnd;
    }

    public getCidadeEnd(): number {
        return this.cidadeEnd;
    }

    public setCidadeEnd(cidadeEnd: number): void {
        this.cidadeEnd = cidadeEnd;
    }

    public getObservacoesEnd(): string {
        return this.observacoesEnd;
    }

    public setObservacoesEnd(observacoesEnd: string): void {
        this.observacoesEnd = observacoesEnd;
    }
}

